src/libs/antares/study/parallel-years.cpp

```diff
@@ -0,0 +1,426 @@
+        numberOfMCYearThreads[ncMax] = 4;
+        break;
+    case 5:
+        numberOfMCYearThreads[ncLow] = 2;
+        numberOfMCYearThreads[ncAvg] = 3;
+        numberOfMCYearThreads[ncHigh] = 4;
+        numberOfMCYearThreads[ncMax] = 5;
+        break;
+    case 6:
+        numberOfMCYearThreads[ncLow] = 2;
+        numberOfMCYearThreads[ncAvg] = 3;
+        numberOfMCYearThreads[ncHigh] = 4;
+        numberOfMCYearThreads[ncMax] = 6;
+        break;
+    case 7:
+        numberOfMCYearThreads[ncLow] = 2;
+        numberOfMCYearThreads[ncAvg] = 3;
+        numberOfMCYearThreads[ncHigh] = 5;
+        numberOfMCYearThreads[ncMax] = 7;
+        break;
+    case 8:
+        numberOfMCYearThreads[ncLow] = 2;
+        numberOfMCYearThreads[ncAvg] = 4;
+        numberOfMCYearThreads[ncHigh] = 6;
+        numberOfMCYearThreads[ncMax] = 8;
+        break;
+    case 9:
+        numberOfMCYearThreads[ncLow] = 3;
+        numberOfMCYearThreads[ncAvg] = 5;
+        numberOfMCYearThreads[ncHigh] = 7;
+        numberOfMCYearThreads[ncMax] = 8;
+        break;
+    case 10:
+        numberOfMCYearThreads[ncLow] = 3;
+        numberOfMCYearThreads[ncAvg] = 5;
+        numberOfMCYearThreads[ncHigh] = 8;
+        numberOfMCYearThreads[ncMax] = 9;
+        break;
+    case 11:
+        numberOfMCYearThreads[ncLow] = 3;
+        numberOfMCYearThreads[ncAvg] = 6;
+        numberOfMCYearThreads[ncHigh] = 8;
+        numberOfMCYearThreads[ncMax] = 10;
+        break;
+    case 12:
+        numberOfMCYearThreads[ncLow] = 3;
+        numberOfMCYearThreads[ncAvg] = 6;
+        numberOfMCYearThreads[ncHigh] = 9;
+        numberOfMCYearThreads[ncMax] = 11;
+        break;
+    default:
+        numberOfMCYearThreads[ncLow] = (uint)std::ceil(nbLogicalCores / 4.);
+        numberOfMCYearThreads[ncAvg] = (uint)std::ceil(nbLogicalCores / 2.);
+        numberOfMCYearThreads[ncHigh] = (uint)std::ceil(3 * nbLogicalCores / 4.);
+        numberOfMCYearThreads[ncMax] = nbLogicalCores - 1;
+        break;
+    }
+
+    /*
+            Getting the number of parallel years based on the number
+            of cores level.
+            This number is limited by the smallest refresh span (if at least
+            one type of time series is generated)
+    */
+
+    try
+    {
+        forcedNbOfParallelYears = numberOfMCYearThreads.at(p.nbCores.ncMode);
+    }
+    catch(const std::out_of_range& e)
+    {
+        logs.fatal() << "Simulation cores level not correct : " << (int)p.nbCores.ncMode;
+    }
+    
+}
+
+
+void SetsOfParallelYearCalculator::limitNbOfParallelYearsbyMinRefreshSpan()
+{
+    uint TSlimit = UINT_MAX;
+    if ((p.timeSeriesToGenerate & timeSeriesLoad) && (p.timeSeriesToRefresh & timeSeriesLoad))
+        TSlimit = p.refreshIntervalLoad;
+    if ((p.timeSeriesToGenerate & timeSeriesSolar) && (p.timeSeriesToRefresh & timeSeriesSolar))
+        TSlimit = (p.refreshIntervalSolar < TSlimit) ? p.refreshIntervalSolar : TSlimit;
```

Comment:
This would be easier to read IMO
```suggestion
        TSlimit = std::min(p.refreshIntervalSolar,  TSlimit);
```
Same remark for the other ones